materialize-bigquery/.snapshots/TestValidateAndApply

```diff
@@ -193,14 +193,14 @@ Big Schema Materialized Resource Schema With No Fields Required:
 
 Big Schema Changed Types With Table Replacement Constraints:
 {"Field":"_meta/flow_truncated","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"Metadata fields fields are able to be materialized"}
-{"Field":"arrayField","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"This field is able to be materialized"}
+{"Field":"arrayField","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"Objects fields may be materialized"}
 {"Field":"boolField","Type":3,"TypeString":"LOCATION_RECOMMENDED","Reason":"The projection has a single scalar type"}
 {"Field":"flow_document","Type":2,"TypeString":"LOCATION_REQUIRED","Reason":"The root document must be materialized"}
 {"Field":"flow_published_at","Type":3,"TypeString":"LOCATION_RECOMMENDED","Reason":"The projection has a single scalar type"}
 {"Field":"intField","Type":3,"TypeString":"LOCATION_RECOMMENDED","Reason":"The projection has a single scalar type"}
 {"Field":"key","Type":2,"TypeString":"LOCATION_REQUIRED","Reason":"All Locations that are part of the collections key are required"}
-{"Field":"multipleField","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"This field is able to be materialized"}
-{"Field":"nullField","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"This field is able to be materialized"}
+{"Field":"multipleField","Type":3,"TypeString":"LOCATION_RECOMMENDED","Reason":"This field is able to be materialized"}
+{"Field":"nullField","Type":4,"TypeString":"FIELD_OPTIONAL","Reason":"Objects fields may be materialized"}
```

Comment:
I find the message here to be confusing since `type: null` is not really an object

Reply:
These snapshots are a little bizarre when viewed in isolation, but may make more sense in the overall sequence of how the tests run. The test that produces these does the following:
1. Applies a materialization spec with a whole bunch of fields, covering every data type I could come up with, where the name of the field corresponds with the type of the field. That is above in `Big Schema Initial Constraints`.
2. Applies the exact same spec again in `Big Schema Re-validated Constraints`.
3. Scrambles the types and tries to apply again in `Big Schema Changed Types Constraints`. In that case, `nullField` was originally `null` and didn't have a column created for it, but the changed types had it become an object, which is now able to be materialized.
4. Does a couple more tests with nullabilty changes to fields in `Big Schema Materialized Resource Schema With All Fields Required` and `Big Schema Materialized Resource Schema With No Fields Required`
5. Finally, the snapshot line in question is produced from the scrambled type snapshot, with the backfill counter incremented on the table. It illustrates that although there were previously incompatible changes, incrementing the backfill counter validates the materialization as-new. With the scrambled type spec, `nullField` is actually an object, which is why it has that `Objects fields may be materialized` (also noting that I need to change the plurality of Objects)